spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/core/convert/StructAccessor.java

```diff
@@ -184,7 +215,8 @@ <T> T getSingleJsonValue(String colName, Class<T> colType) {
     return gson.fromJson(jsonString, colType);
   }
 
-  public <T> T getSingleJsonValue(int colIndex, Class<T> colType) {
+  //TODO: change this to private in next major release
```

Comment:
Yes, `getJsonValue` is public and `getSingleJsonValue` and `getListJsonValue` does not need to be exposed. In fact, this method is mostly for internal usage. Users don’t need to interact with neither of these methods, as shown in the samples, user only need to add annotation of the dedicated field and use save()/find() as usual. But this method needs to be public because it is used outside of its package.